greenheart/simulation/technologies/hydrogen/h2_transport/h2_export_pipe.py

```diff
@@ -2,167 +2,296 @@
 
-def get_anl_costs(costs,total_L):
-    labor_coef = [95295,0.53848,0.03070]
-    misc_coef = [19211,0.14178,0.04697]
-    row_coef = [72634,1.07566,0.05284]
-    mat_coef = [5605,0.41642,0.06441] # Don't need this anymore since material cost is coming from Savoy numbers
-
-    L_mi = total_L*0.621371
 
-    # costs['mat cost anl [$]'] = costs.apply(lambda x: (mat_coef[0]*((x['Outer diameter (mm)']*mm2in)**mat_coef[1])/L_mi**mat_coef[2])*(x['Outer diameter (mm)']*mm2in*L_mi),axis=1)
-    costs['labor cost [$]'] = costs.apply(lambda x: (labor_coef[0]/((x['Outer diameter (mm)']*mm2in)**labor_coef[1])*L_mi**labor_coef[2])*(x['Outer diameter (mm)']*mm2in*L_mi),axis=1)
-    costs['misc cost [$]'] = costs.apply(lambda x: (misc_coef[0]/((x['Outer diameter (mm)']*mm2in)**misc_coef[1])/L_mi**misc_coef[2])*(x['Outer diameter (mm)']*mm2in*L_mi),axis=1)
-    costs['ROW cost [$]'] = costs.apply(lambda x: (row_coef[0]/((x['Outer diameter (mm)']*mm2in)**row_coef[1])*L_mi**row_coef[2])*(x['Outer diameter (mm)']*mm2in*L_mi),axis=1)
+def get_anl_costs(costs: pd.DataFrame, total_L: float) -> pd.DataFrame:
+    labor_coef = [95295, 0.53848, 0.03070]
+    misc_coef = [19211, 0.14178, 0.04697]
+    row_coef = [72634, 1.07566, 0.05284]
+    mat_coef = [
+        5605,
+        0.41642,
+        0.06441,
+    ]  # Don't need this anymore since material cost is coming from Savoy numbers
 
-    costs['total capital cost [$]'] = costs[['mat cost [$]','labor cost [$]','misc cost [$]','ROW cost [$]']].sum(axis=1)
+    L_mi = total_L * 0.621371
 
-    costs['annual operating cost [$]'] = 0.0117*costs['total capital cost [$]'] # https://doi.org/10.1016/j.esr.2021.100658
+    # costs['mat cost anl [$]'] = costs.apply(lambda x: (mat_coef[0]*((x['Outer diameter (mm)']*mm2in)**mat_coef[1])/L_mi**mat_coef[2])*(x['Outer diameter (mm)']*mm2in*L_mi),axis=1)
+    costs["labor cost [$]"] = costs.apply(
+        lambda x: (
+            labor_coef[0]
+            / ((x["Outer diameter (mm)"] * MM2IN) ** labor_coef[1])
+            * L_mi ** labor_coef[2]
+        )
+        * (x["Outer diameter (mm)"] * MM2IN * L_mi),
+        axis=1,
+    )
+    costs["misc cost [$]"] = costs.apply(
+        lambda x: (
+            misc_coef[0]
+            / ((x["Outer diameter (mm)"] * MM2IN) ** misc_coef[1])
+            / L_mi ** misc_coef[2]
+        )
+        * (x["Outer diameter (mm)"] * MM2IN * L_mi),
+        axis=1,
+    )
+    costs["ROW cost [$]"] = costs.apply(
+        lambda x: (
+            row_coef[0]
+            / ((x["Outer diameter (mm)"] * MM2IN) ** row_coef[1])
+            * L_mi ** row_coef[2]
+        )
+        * (x["Outer diameter (mm)"] * MM2IN * L_mi),
+        axis=1,
+    )
+
+    costs["total capital cost [$]"] = costs[
+        ["mat cost [$]", "labor cost [$]", "misc cost [$]", "ROW cost [$]"]
+    ].sum(axis=1)
+
+    costs["annual operating cost [$]"] = (
+        0.0117 * costs["total capital cost [$]"]
+    )  # https://doi.org/10.1016/j.esr.2021.100658
```

Comment:
Does this reference apply to the whole function? If so, can we move it to a doc string?